File: packages/react-native-autoplay/ios/hybrid/HybridGridTemplate.swift
//
//  HybridGridTemplate.swift
//  Pods
//
//  Created by Manuel Auer on 15.10.25.
//

import NitroModules

class HybridGridTemplate: HybridGridTemplateSpec {
    func createGridTemplate(config: GridTemplateConfig) throws {
        let template = GridTemplate(config: config)

        TemplateStore.addTemplate(
            template: template,
            templateId: config.id
        )
    }

    func updateGridTemplateButtons(
        templateId: String,
        buttons: [NitroGridButton]
    ) throws -> Promise<Void> {
        return Promise.async {
            guard
                let template = TemplateStore.getTemplate(templateId: templateId)
                    as? GridTemplate
            else {
                throw AutoPlayError.templateNotFound(templateId)
            }

            await template.updateButtons(buttons: buttons)
        }
    }
}

File: packages/react-native-autoplay/android/src/main/java/com/margelo/nitro/swe/iternio/reactnativeautoplay/HybridGridTemplate.kt
package com.margelo.nitro.swe.iternio.reactnativeautoplay

import com.margelo.nitro.core.Promise
import com.margelo.nitro.swe.iternio.reactnativeautoplay.template.AndroidAutoTemplate
import com.margelo.nitro.swe.iternio.reactnativeautoplay.template.GridTemplate

class HybridGridTemplate : HybridGridTemplateSpec() {
    override fun createGridTemplate(config: GridTemplateConfig) {
        val context = AndroidAutoSession.getRootContext()
            ?: throw IllegalArgumentException("createGridTemplate failed, carContext found")

        val template = GridTemplate(context, config)
        AndroidAutoTemplate.setTemplate(config.id, template)
    }

    override fun updateGridTemplateButtons(
        templateId: String, buttons: Array<NitroGridButton>
    ): Promise<Unit> {
        return Promise.async {
            val template = AndroidAutoTemplate.getTemplate<GridTemplate>(templateId)
            template.updateButtons(buttons)
        }
    }
}